File: booking website(reference)/request.js
class torequest{
    constructor(date, time, type, amount, name, email, phone){
        this.person = name
        this.date = date
        this.time = time
        this.type = type
        this.amount = amount
        this.email = email
        this.phone = phone
    }

}
let fullname = window.prompt("What is your name? ")
let date = window.prompt("What date do you want to go? ")
let type = window.prompt("What location do you want to go? ")
let amount = window.prompt("What is the amount people that are going")
let time = window.prompt("What time do you want to go? ")
let email = window.prompt("What is your email? ")
let phone = window.prompt("What is you phone number? ")

let newrequest = new torequest(fullname,date, type, amount, time, email, phone)

fs.writeFile('tourSchedule.json', newrequest, (err) => {
    if (err) {
        throw err;
    }
    console.log("JSON data is saved.");
});

File: booking website(reference)/request.cpp
#include <iostream>
#include <vector>

#include "request.h"

request::request(string date, string time, string type, string amount, string name, string email, string phone){
    this->date = date;
    array.push_back(date);
    this->time = time;
    array.push_back(time);
    this->type = type;
    array.push_back(type);
    this->amount = amount;
    array.push_back(amount);
    this->name = name;
    array.push_back(name);
    this->email = email;
    array.push_back(email);
    this-> phone = phone;
    array.push_back(phone);
}

request::~request(){
    for(int i = 0; i < array.size(); i++){
        array.erase(array.begin());
    }
}
vector<string> request::getArray(){
    return array;
}




